integration-tests/src/tests/client/resharding.rs

```diff
@@ -190,10 +191,14 @@ impl TestReshardingEnv {
             genesis_protocol_version,
         );
         let chain_genesis = ChainGenesis::new(&genesis);
-        let env = TestEnv::builder(chain_genesis)
+        let builder = if state_snapshot_enabled {
```

Comment:
This is super cool as now we can run the resharding tests without enabling snapshot. For testing purposes, I have tests for both when the snapshot config is enabled and disabled.